pkg/operator/k8sutil/configmap.go

```diff
@@ -62,13 +64,32 @@ func GetOperatorSetting(context context.Context, clientset kubernetes.Interface,
 }
 
 func GetValue(data map[string]string, settingName, defaultValue string) string {
-	if settingValue, ok := data[settingName]; ok {
-		logger.Infof("%s=%q (configmap)", settingName, settingValue)
-		return settingValue
-	} else if settingValue, ok := os.LookupEnv(settingName); ok {
-		logger.Infof("%s=%q (env var)", settingName, settingValue)
-		return settingValue
+	var settingValue, settingSource string
+
+	if val, ok := data[settingName]; ok {
+		settingValue = val
+		settingSource = "configmap"
+	} else if val, ok := os.LookupEnv(settingName); ok {
+		settingValue = val
+		settingSource = "env var"
+	} else {
+		settingValue = defaultValue
+		settingSource = "default"
+	}
+
+	if settingValue != "" {
```

Comment:
It's also interesting to log a blank setting, right?
```suggestion
```

Reply:
yeah, it will not harm us, updated the code